### PR TITLE
Jbuilder

### PR DESCRIPTION
**Pull request by vbmithr**
Initially wanted to add set/get functions for various integers, but found out that `ocplib-endian` is already perfectly suited for that use case.

### CODE REVIEW

File: bigstring.opam [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+name: "bigstring"
+version: "0.1.1"
+authors: "Simon Cruanes <simon.cruanes.2007@m4x.org>"
+maintainer: "Simon Cruanes <simon.cruanes.2007@m4x.org>"
+tags: [ "bigstring" "bigarray" ]
+homepage: "https://github.com/c-cube/ocaml-bigstring/"
+bug-reports: "https://github.com/c-cube/ocaml-bigstring/issues"
+dev-repo: "git://github.com/c-cube/ocaml-bigstring"
+
+available: [ ocaml-version >= "4.06.0" ]
```
Comment (c-cube) [inline bigstring.opam]: why such a constraint? This makes the library unusable for anyone except people living on the bleeding edge

File: src/bigstring.ml [hunk 2]
```diff
@@ -395,7 +395,8 @@ let lines s = split ~by:'\n' s
 (** {2 Memory-map} *)
 
 let map_file_descr ?pos ?(shared=false) fd len =
-  B.map_file fd ?pos Bigarray.char Bigarray.c_layout shared len
+  Bigarray.array1_of_genarray
```
Comment (c-cube) [inline src/bigstring.ml]: I suppose we should move this to bigstring.unix or something like that, with an explicit dep on unix (and some jbuilder magic to stil work pre-4.06)

**Comment by c-cube** [comment]
Thank you for the contribution! :slightly_smiling_face: 
I think it's a good idea to move to jbuilder, but not if it requires >= 4.06 (although I suppose that comes from the warning part)

File: src/bigstring.ml (hunk 2 above)
Comment (vbmithr) [reply src/bigstring.ml]: I suggest to cancel this change and just ignore the warning…

Comment (c-cube) [reply src/bigstring.ml]: that sounds like a good idea indeed :-)
do you want to do it, or should I do it after merging?

**Comment by vbmithr** [comment]
I finally fixed pre-4.03 warnings and moved memory mapping to a separate opam package.

File: src/jbuild [hunk 3]
```diff
@@ -0,0 +1,15 @@
+(jbuild_version 1)
+
+(library
+ ((name        bigstring)
+  (public_name bigstring)
```
Comment (c-cube) [inline src/jbuild:5]: don't we want `(wrapped false)` in both libraries, btw?

edit: nevermind, I forgot it worked if the module name matched the lib's name

File: src/jbuild [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  (modules (Bigstring))
+  (synopsis "Bigstring built on top of bigarrays, and convenient functions.")
+  (libraries (bytes bigarray))))
+
+(library
```
Comment (c-cube) [inline src/jbuild:10]: :+1:

**Comment by c-cube** [comment]
thank you! :smile:

**Pull request merged**